bridges/WikipediaBridge.php

```diff
@@ -107,9 +107,18 @@ private function AddElementGeneric($element, $fullArticle){
 		// Clean the bottom of the featured article
 		$element->find('div', -1)->outertext = '';
 
+		// The title and URI of the article is best defined in an anchor containint the string '...' ('full article ...')
+		$target = $element->find('p/a', 0); // We'll use the first anchor as fallback
+		foreach($element->find('//a') as $anchor){
+			if(strpos($anchor->innertext, '...') !== false){
```

Comment:
Great idea making this whole thing into a more generic framework!

I think a better approach would be to match against an array of various "read more" texts with "..." as a kind of generic read-more fallback before the first link fallback. I admit that my own approach in the French Wikipedia was overly complicated by matching against different classnames and simply having matched the text against "Lire la suite" would've simpler. In any case, this algorithm would've broken down on 14 August for French Wikipedia, and I'm sure it'd break down for Dutch and Italian Wikipedia at some point as well. That being said, a quick look around suggests that those examples seem to be somewhat exceptional in not using "..."

Anyway, off the top of my head, how about something like this?
```php
function strpos_in_array($word, $array) {
	foreach($array as $a) {
		if (strpos($word, $a) !== false) {
			return true;
		}
	}
	return false;
}
```

The array would need to contain something like

```php
$read_more_text = array(
	'Lees verder', //Dutch
	'Leggi la voce', //Italian
	'Lire la suite', //French
	'...' //Generic
);
```

I can also toss it into a PR myself later if you prefer.